tools/idf.py

```diff
@@ -713,7 +713,32 @@ def main() -> None:
         else:
             raise
     else:
-        cli(sys.argv[1:], prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+        argv = expand_file_arguments(argv or sys.argv[1:])
+
+        cli(argv, prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+
+
+def expand_file_arguments(argv):
+    """
+    Any argument starting with "@" gets replaced with all values read from a text file.
+    Text file arguments can be split by newline or by space.
+    Values are added "as-is", as if they were specified in this order
+    on the command line.
+    """
+    new_args = []
+    expanded = False
+    for arg in argv:
+        if arg.startswith("@"):
+            expanded = True
+            with open(arg[1:], "r") as f:
```

Comment:
I would suggest printing a meaningful error for the user in the case when the file doesn't exist instead of python stack trace.